WhitespaceAfter should support `ELLIPSIS` token

I have downloaded the latest cli from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

Check documentation at https://checkstyle.sourceforge.io/config_whitespace.html#WhitespaceAfter

**How it works Now:**

```bash

➜  javac Test.java
➜  cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
        <module name="WhitespaceAfter">
          <property name="tokens" value="ELLIPSIS"/>
        </module>
    </module>
</module>

➜  cat Test.java  
public class Test {
    void method1(int...myVarargs) { // expected violation
    
    }
    void method2(int... myVarargs) { // ok
    
    }
}

➜  java -jar checkstyle-9.2-all.jar -c config.xml Test.java
com.puppycrawl.tools.checkstyle.api.CheckstyleException: cannot initialize module TreeWalker - Token "ELLIPSIS" was not found in Acceptable tokens list in check com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck
        at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:473)
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:201)
        at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:405)
        at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:332)
        at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:191)
        at com.puppycrawl.tools.checkstyle.Main.main(Main.java:126)
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Token "ELLIPSIS" was not found in Acceptable tokens list in check com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck
        at com.puppycrawl.tools.checkstyle.TreeWalker.registerCheck(TreeWalker.java:223)
        at com.puppycrawl.tools.checkstyle.TreeWalker.setupChild(TreeWalker.java:133)
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:201)
        at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:468)
        ... 5 more
Checkstyle ends with 1 errors.

```

However, `WhitespaceAroundCheck` does support `ELLIPSIS`. But, this is not a frequently used style. In fact, the [JLS](https://docs.oracle.com/javase/specs/jls/se8/html/jls-8.html) specifically mentions that

> Note that the ellipsis (...) is a token unto itself (§3.11). It is possible to put whitespace between it and the type, but this is discouraged as a matter of style.

**Is your feature request related to a problem? Please describe.**
I would expect `...` to be a valid token for `WhitespaceAfterCheck`, and for it to enforce style noted above.

**Describe the solution you'd like**


```bash
[ERROR] Test.java:2:21: '...' is not followed by whitespace. [WhitespaceAfter]
```

**Additional context**

Noticed at https://github.com/checkstyle/checkstyle/pull/11094#discussion_r775249437, WhitespaceAfterCheck does not currently support `ELLIPSIS` token.

`ELLIPSIS` should be a default token for `WhitespaceAfterCheck`. Since we use default module for this check in `checkstyle_checks.xml`, we may find some violations in the project.

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java
@@ -55,7 +55,9 @@ import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
  * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_FOR">
  * LITERAL_FOR</a>,
  * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_DO">
- * DO_WHILE</a>.
+ * DO_WHILE</a>,
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ELLIPSIS">
+ * ELLIPSIS</a>.
  * </li>
  * </ul>
  * <p>
@@ -150,6 +152,7 @@ public class WhitespaceAfterCheck
             TokenTypes.LITERAL_DO,
             TokenTypes.LITERAL_FOR,
             TokenTypes.DO_WHILE,
+            TokenTypes.ELLIPSIS,
         };
     }
 

--- a/src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks/whitespace/WhitespaceAfterCheck.xml
+++ b/src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks/whitespace/WhitespaceAfterCheck.xml
@@ -11,7 +11,7 @@
  EmptyForIteratorPad&lt;/a&gt; to validate empty for iterators.
  &lt;/p&gt;</description>
          <properties>
-            <property default-value="COMMA,SEMI,TYPECAST,LITERAL_IF,LITERAL_ELSE,LITERAL_WHILE,LITERAL_DO,LITERAL_FOR,DO_WHILE"
+            <property default-value="COMMA,SEMI,TYPECAST,LITERAL_IF,LITERAL_ELSE,LITERAL_WHILE,LITERAL_DO,LITERAL_FOR,DO_WHILE,ELLIPSIS"
                       name="tokens"
                       type="java.lang.String[]"
                       validation-type="tokenSet">

--- a/src/main/resources/google_checks.xml
+++ b/src/main/resources/google_checks.xml
@@ -105,7 +105,7 @@
     <module name="WhitespaceAfter">
       <property name="tokens"
                value="COMMA, SEMI, TYPECAST, LITERAL_IF, LITERAL_ELSE,
-                    LITERAL_WHILE, LITERAL_DO, LITERAL_FOR, DO_WHILE"/>
+                    LITERAL_WHILE, LITERAL_DO, LITERAL_FOR, DO_WHILE, ELLIPSIS"/>
     </module>
     <module name="WhitespaceAround">
       <property name="allowEmptyConstructors" value="true"/>

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheckTest.java
@@ -186,4 +186,16 @@ public class WhitespaceAfterCheckTest
         verifyWithInlineConfigParser(
                 getPath("InputWhitespaceAfterCountUnicodeCorrectly.java"), expected);
     }
+
+    @Test
+    public void testVarargs() throws Exception {
+        final String[] expected = {
+            "14:27: " + getCheckMessage(MSG_WS_NOT_FOLLOWED, "..."),
+            "17:67: " + getCheckMessage(MSG_WS_NOT_FOLLOWED, "..."),
+            "20:42: " + getCheckMessage(MSG_WS_NOT_FOLLOWED, "..."),
+            "27:45: " + getCheckMessage(MSG_WS_NOT_FOLLOWED, "..."),
+            "36:19: " + getCheckMessage(MSG_WS_NOT_FOLLOWED, "..."),
+        };
+        verifyWithInlineConfigParser(getPath("InputWhitespaceAfterVarargs.java"), expected);
+    }
 }

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterVarargs.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterVarargs.java
@@ -1,0 +1,37 @@
+/*
+WhitespaceAfter
+tokens = ELLIPSIS
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.whitespace.whitespaceafter;
+
+import java.util.List;
+
+public class InputWhitespaceAfterVarargs {
+
+    public void method(int...obj) { // violation
+    }
+
+    public InputWhitespaceAfterVarargs(String first, List<Integer>...second) { // violation
+    }
+
+    public <T> void anotherMethod(List<T>...args) { // violation
+    }
+
+    public static String multipleArguments(int l, String format, Object ... args) { // ok
+        return format;
+    }
+
+    private void noWhitespaceBefore(boolean ...args) { // violation
+    }
+
+    testInterface<Integer> obj = (Integer... i) -> { // ok
+    };
+}
+
+@FunctionalInterface
+interface testInterface<T> {
+    void method(T ...args); // violation
+}

--- a/src/xdocs/config_whitespace.xml
+++ b/src/xdocs/config_whitespace.xml
@@ -2902,6 +2902,8 @@ class Bar {
                   LITERAL_FOR</a>
                 , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_DO">
                   DO_WHILE</a>
+                , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ELLIPSIS">
+                  ELLIPSIS</a>
                   .
               </td>
 
@@ -2924,6 +2926,8 @@ class Bar {
                   LITERAL_FOR</a>
                 , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_DO">
                   DO_WHILE</a>
+                , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ELLIPSIS">
+                  ELLIPSIS</a>
                   .
               </td>
               <td>3.0</td>

